### DISCUSSION TITLE
Vue variables as variants?

### DISCUSSION THREAD

**Discussion by itzaks** [unocss/unocss#920] [Ideas]
Hi!

I'm curious if there would be a way to write variant rules so to match vue component variables. Because right now I'm writing conditional stuff like this (pug):

````pug
div(:flex="growContent ? '~ row grow' : '~ row'")
````

Where `growContent` is a prop on the vue component.

I would love to be able to write it like:

````pug
div(flex="~ row grow-content:grow")
````

**Comment by lunelson** [comment]
IMO the best way to marry runtime style variation and build-time CSS is through CSS custom properties (variables)—you could write a function to make the custom-property generation terser but the essential idea would be this

```pug
div(:flex="row grow", style={'--grow': growContent ? 0 : 1 })
```
```css
.grow {
  flex-grow: var(--grow, 1);
}
```
